File: flutter/platform/lib/src/modules/legal.dart
import '../core/http_client.dart';
import '../core/api_endpoints.dart';
import '../core/result.dart';

/// Legal module — documents and acceptance tracking.
class LegalModule {
  const LegalModule();

  /// Get all legal documents (public, no auth required).
  Future<ApiResult<List<dynamic>>> getDocuments() =>
      HttpClient.instance.getList(ApiEndpoints.legalDocuments);

  /// Get a single legal document by type (privacy, terms, cookie, refund).
  Future<ApiResult<Map<String, dynamic>>> getDocument(String type) =>
      HttpClient.instance.getMap(ApiEndpoints.legalDocumentByType(type));

  /// Accept a legal document version.
  Future<ApiResult<Map<String, dynamic>>> accept(
    String documentType,
    String version,
  ) => HttpClient.instance.postMap(ApiEndpoints.legalAccept, body: {
    'documentType': documentType,
    'version': version,
  });

  /// Check acceptance status for all documents.
  Future<ApiResult<Map<String, dynamic>>> getAcceptanceStatus() =>
      HttpClient.instance.getMap(ApiEndpoints.legalAcceptanceStatus);
}

File: js/platform/src/modules/legal.ts
import { http } from "../core/http.js";
import { Endpoints } from "../core/endpoints.js";
import type { ApiResult } from "../core/result.js";

export const legal = {
  getDocuments(): Promise<ApiResult<unknown[]>> {
    return http.getList(Endpoints.legalDocuments);
  },
  getDocument(type: string): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.legalDocumentByType(type));
  },
  accept(documentType: string, version: string): Promise<ApiResult<Record<string, unknown>>> {
    return http.postMap(Endpoints.legalAccept, { documentType, version });
  },
  getAcceptanceStatus(): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.legalAcceptanceStatus);
  },
};
